pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -167,6 +167,30 @@ func handleError(message string, listOfErrors ...error) error {
 	return fmt.Errorf("errors: %s, msg: %s", strings.Join(errSlice, ", "), message)
 }
 
+// GetBuildRunObject retrieves an existing BuildRun based on a name and namespace
+func (r *ReconcileBuildRun) GetBuildRunObject(ctx context.Context, objectName string, objectNS string) (*buildv1alpha1.BuildRun, error) {
+	var buildRun buildv1alpha1.BuildRun
+	if err := r.client.Get(ctx, types.NamespacedName{Name: objectName, Namespace: objectNS}, &buildRun); err != nil {
+		return nil, err
+	}
+	return &buildRun, nil
+}
+
+// GetBuildObject retrieves an existing Build based on a name and namespace
+func (r *ReconcileBuildRun) GetBuildObject(ctx context.Context, objectName string, objectNS string, buildRun *buildv1alpha1.BuildRun) (*buildv1alpha1.Build, error) {
+	var build buildv1alpha1.Build
+	if err := r.client.Get(ctx, types.NamespacedName{Name: objectName, Namespace: objectNS}, &build); err != nil {
+		return nil, err
+	}
+
+	if build.Status.Registered != corev1.ConditionTrue {
+		err := fmt.Errorf("The Build is not registered correctly, registered status: %s, reason: %s", build.Status.Registered, build.Status.Reason)
+		updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
+		return nil, handleError("Build is not ready", err, updateErr)
+	}
+	return &build, nil
+}
```

Comment:
This is a `get` function with a side-effect (updating the buildrun in case of status registered != true). A get function should not have such side-effects.

Reply:
thanks, fixed